### PR TITLE
Warning fixes and TSG abort condition fix

### PR DESCRIPTION
**Pull request by akallabeth**
Fixes warnings occuring on current master.

### CODE REVIEW

File: libfreerdp/core/gateway/http.c [hunk 1]
```diff
@@ -362,7 +362,7 @@ static BOOL http_encode_print(wStream* s, const char* fmt, ...)
 {
 	char* str;
 	va_list ap;
-	size_t length, used;
+	int length, used;
```
Comment (hardening) [inline libfreerdp/core/gateway/http.c]: I don't understand what is done here, you change these variables to `int` and then they are cast to `size_t` in the rest of the function. What was wrong before ?

Comment (akallabeth) [reply libfreerdp/core/gateway/http.c]: @hardening The return value of `vsnprintf` is int, but other functions require `size_t` as argument.

Comment (hardening) [reply libfreerdp/core/gateway/http.c]: Then perhaps just casting the calls to `vsnprintf` would introduce less changes and would improve readability ?

Comment (akallabeth) [reply libfreerdp/core/gateway/http.c]: it is a return value of `vsnprintf`...

File: libfreerdp/core/gateway/tsg.c [hunk 2]
```diff
@@ -670,14 +676,7 @@ static BOOL TsProxyCreateTunnelReadResponse(rdpTsg* tsg, RPC_PDU* pdu,
...
 			case TSG_ASYNC_MESSAGE_REAUTH:
 				{
-					size_t loffset = Stream_GetPosition(pdu->s);
-					UINT32 offset = (UINT32)loffset;
-
-					if (loffset > UINT32_MAX)
-						goto fail;
-
-					if (!Stream_SafeSeek(pdu->s, rpc_offset_align(&offset, 8) ||
-					                     (Stream_GetRemainingLength(pdu->s) < 8)))
```
Comment (nfedera) [inline libfreerdp/core/gateway/tsg.c]: - Don't know the internals but the new tsg_stream_align always uses `4` while we had `8`.
- Not sure if the removed checks were important

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**